packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -329,13 +329,15 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
     assert(() {
       if (debugPaintSizeEnabled) {
         final Size parentSize = getAbsoluteSizeRelativeToOrigin();
-        final Rect outerRect = offset & parentSize;
+        // offset is absolute coordinate, we will need to convert relative
+        // size to absolute size.
+        final Rect outerRect = offset & Size(parentSize.width.abs(), parentSize.height.abs());
```

Comment:
I wonder if we should add a protected method "getAbsoluteSize" that just returns the correct absolute size directly without calls to abs().